src/3p-analytics-common.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** @const {string} */
+export const AMP_ANALYTICS_3P_EVENT_MESSAGES_TYPE = 'AA3pEvtMsgs';
+
+/** @typedef {!Object<!string,!Array<!string>>} */
+export let AmpAnalytics3pEventMap;
```

Comment:
1) since we are talking about a continuously data streaming, an array like structure is  more flexible and intuitive.
2) the 3p code should not care about the wire format, our lib should take care of it and convert to an API like this:

```
window.context.onAnalyticsEvent(event => {
  // handle the IframeTransportEvent 
});
```
we hide the batching / throttling details from the client. we will make sure the `callback` is invoked for every event arrived at client

that's said, there is no need to prepare a map or so. in fact, i don't even know if the `transportId` means anything at their side. they should have had the creativeId or so in the message.